crates/pow/Cargo.toml

```diff
@@ -15,10 +15,10 @@ version.workspace      = true
 workspace = true
 
 [dependencies]
-serde     = { workspace = true }
-sha2      = { workspace = true }
-thiserror = { workspace = true }
-tokio     = { features = ["fs"], workspace = true }
+miden-client = { workspace = true }
```

Comment:
Could we avoid introducing `miden-client` dependency here? I think `Challenge` struct is simple enough where we can probably serialize it "by hand" and it may actually end up being less code. For example, we could have something like:

```rust
impl Challenge {
    pub fn to_bytes(&self) -> [u8; SERIALIZED_SIZE] {
        ...
    }
}

impl From<Challenge> for [u8;  SERIALIZED_SIZE] {
    ...
}

impl TryFrom<[u8; SERIALIZED_SIZE]> for Challenge {
    ...
}

impl TryFrom<&[u8]> for Challenge {
    ...
}
```